XamarinCommunityToolkit/Behaviors/BaseBehavior.shared.cs

```diff
@@ -1,74 +1,63 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Text;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
 using Xamarin.Forms;
-using Xamarin.Forms.Internals;
-using XamarinCommunityToolkit.Controls;
 
 namespace XamarinCommunityToolkit.Behaviors
 {
-    /// <summary>
-    /// Contains base functionality for all derived behaviors
-    /// </summary>
-    /// <typeparam name="T"></typeparam>
-    [Preserve(AllMembers = true)]
-    public class BaseBehavior<T> : Behavior<T> where T : BindableObject
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public abstract class BaseBehavior : Behavior<View>
     {
-        public static readonly BindableProperty ActionsProperty =
-            BindableProperty.Create(nameof(Actions), typeof(ActionCollection), typeof(BaseBehavior<BindableObject>), null);
+        static readonly MethodInfo getContextMethod = typeof(BindableObject).GetRuntimeMethods()?.FirstOrDefault(m => m.Name == "GetContext");
```

Comment:
I'm assuming there is no other way than doing it like this? If we can make a case for why we want this, we could open a dialog with the Forms team to see if they can provide us with a way to access this more easily?

Reply:
Yes, we cannot do it right now by normal approach cuz existing method **GetIsBound** is **internal** and even isn't what we need. That method checks property binding (if it exists), but we also need to check if this binding is ours or not.

So, I use this method for checking whether we need to set the default binding to some properties or not (BindingContext for every BaseBehavior & Value for ValidationBehavior)